microbench/src/main/java/io/netty/buffer/AbstractReferenceCountedByteBufBenchmark.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.buffer;
+
+import io.netty.microbench.util.AbstractMicrobenchmark;
+import org.openjdk.jmh.annotations.*;
+import org.openjdk.jmh.infra.Blackhole;
+
+import java.util.concurrent.TimeUnit;
+
+@State(Scope.Benchmark)
+public class AbstractReferenceCountedByteBufBenchmark extends AbstractMicrobenchmark {
+
+    @Param({"1", "10", "100", "1000", "10000"})
+    public int delay;
+
+    AbstractReferenceCountedByteBuf buf;
+
+    @Setup
+    public void setUp() {
+        buf = (AbstractReferenceCountedByteBuf) Unpooled.buffer(1);
```

Comment:
why you even need t do any casing at all ?

Reply:
Unpooled.buffer returns (ByteBuf).  The API doesn't assert that it will always be a AbstractReferenceCountedByteBuf , so I cast here so that the benchmark fails if this assumption changes.